### PR TITLE
fix: Only generating `VarCompletion` if it matches prefix

### PR DESCRIPTION
**Pull request by LukasRonn**

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/api/lsp/provider/completion/VarCompleter.scala [hunk 1]
```diff
@@ -32,6 +33,20 @@ object VarCompleter extends Completer {
       case Entity.FormalParam(fparam) => Completion.VarCompletion(fparam.sym, fparam.tpe)
     }
 
-    iter.toList
+    iter.toList.filter(comp => matchesVar(comp.sym, context.word))
+  }
+
+  /**
+    * Checks that the varSym matches the prefix
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/api/lsp/provider/completion/VarCompleter.scala:40]: What prefix?

File: main/src/ca/uwaterloo/flix/api/lsp/provider/completion/VarCompleter.scala [hunk 2]
```diff
@@ -32,6 +33,20 @@ object VarCompleter extends Completer {
...
+    *
+    * @param sym  the varSym.
+    * @param word the current prefix.
+    * @return     true, if the var matches prefix, false otherwise.
+    */
+  private def matchesVar(sym: Symbol.VarSym, word: String): Boolean = {
+    if (word.nonEmpty && word.head.isUpper)
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/api/lsp/provider/completion/VarCompleter.scala:47]: Variables cannot be uppercase (??)

**Comment by magnus-madsen** [comment]
What does this change do?

File: main/src/ca/uwaterloo/flix/api/lsp/provider/completion/VarCompleter.scala (hunk 1 above)
Comment (LukasRonn) [reply main/src/ca/uwaterloo/flix/api/lsp/provider/completion/VarCompleter.scala:40]: The prefix the user has typed in VSCode

File: main/src/ca/uwaterloo/flix/api/lsp/provider/completion/VarCompleter.scala (hunk 2 above)
Comment (LukasRonn) [reply main/src/ca/uwaterloo/flix/api/lsp/provider/completion/VarCompleter.scala:47]: I was a bit too quick there. That is indeed true.

**Comment by LukasRonn** [comment]
> What does this change do?

This ensures that `VarCompletion`s if the var match the word that the user is currently typing.

**Comment by LukasRonn** [comment]
Example:

```
def bar(): String = 
   let oldString = "Hello";
   let newString = "Hello World!";
   n..
```

When you make a request to autocomplete with the prefix `n`, we are actually also creating a `VarCompletion` for `oldString`, even though it doesn't match. This is generally a problem for ranking `Completion`, if we create some `Completion` that doesn't match the prefix the user is typing. When that is the case, these `Completion` can overshadow the ones we actually need (those who matches prefix) and result in a wrongful boosting of a `Completion` in `CompletionRanker.scala`.

In this case, we won't get a "Best Pick", because `oldString` overshadows, because it appears before `newString` in the list of all possible completions.

**Pull request merged**